cmake/fake_tribits.cmake

```diff
@@ -34,88 +32,59 @@ MACRO(ADD_INTERFACE_LIBRARY LIB_NAME)
+  #  * `-D <fullTestName>_SET_RUN_SERIAL=ON`
+  IF(${TEST_NAME}_DISABLE)
+    RETURN()
+  ENDIF()
 
-    # The function TRIBITS_ADD_TEST() has a CATEGORIES argument that defaults
-    # to BASIC.  If a project elects to only enable tests marked as PERFORMANCE,
-    # the test won't actually be added and attempting to set a property on it below
-    # will yield an error.
-    if(TARGET ${EXE})
-      if(TEST_TOOL)
-        add_dependencies(${EXE} ${TEST_TOOL}) #make sure the exe has to build the tool
-        foreach(TEST_ADDED ${ALL_TESTS_ADDED})
-          set_property(TEST ${TEST_ADDED} APPEND PROPERTY ENVIRONMENT "KOKKOS_TOOLS_LIBS=$<TARGET_FILE:${TEST_TOOL}>")
-        endforeach()
-      endif()
-    endif()
-  else()
-    CMAKE_PARSE_ARGUMENTS(TEST
-      "WILL_FAIL;SKIP_TRIBITS"
-      "FAIL_REGULAR_EXPRESSION;PASS_REGULAR_EXPRESSION;EXE;NAME;TOOL"
-      "CATEGORIES;ARGS"
-      ${ARGN})
-    # To match Tribits, we should always be receiving
-    # the root names of exes/libs
-    IF(TEST_EXE)
-      SET(EXE_ROOT ${TEST_EXE})
-    ELSE()
-      SET(EXE_ROOT ${TEST_NAME})
-    ENDIF()
-    # Prepend package name to the test name
-    # These should be the full target name
-    SET(TEST_NAME ${PACKAGE_NAME}_${TEST_NAME})
-    SET(EXE ${PACKAGE_NAME}_${EXE_ROOT})
-    IF(WIN32)
-      ADD_TEST(NAME ${TEST_NAME} WORKING_DIRECTORY ${LIBRARY_OUTPUT_PATH}
-        COMMAND ${EXE}${CMAKE_EXECUTABLE_SUFFIX} ${TEST_ARGS})
-    ELSE()
-      ADD_TEST(NAME ${TEST_NAME} COMMAND ${EXE} ${TEST_ARGS})
-    ENDIF()
-    IF(TEST_WILL_FAIL)
-      SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES WILL_FAIL ${TEST_WILL_FAIL})
-    ENDIF()
-    IF(TEST_FAIL_REGULAR_EXPRESSION)
-      SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES FAIL_REGULAR_EXPRESSION ${TEST_FAIL_REGULAR_EXPRESSION})
-    ENDIF()
-    IF(TEST_PASS_REGULAR_EXPRESSION)
-      SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES PASS_REGULAR_EXPRESSION ${TEST_PASS_REGULAR_EXPRESSION})
-    ENDIF()
-    IF(TEST_TOOL)
-      ADD_DEPENDENCIES(${EXE} ${TEST_TOOL}) #make sure the exe has to build the tool
-      SET_PROPERTY(TEST ${TEST_NAME} APPEND_STRING PROPERTY ENVIRONMENT "KOKKOS_PROFILE_LIBRARY=$<TARGET_FILE:${TEST_TOOL}>")
-    ENDIF()
-    VERIFY_EMPTY(KOKKOS_ADD_TEST ${TEST_UNPARSED_ARGUMENTS})
+  SET(EXE ${PACKAGE_NAME}_${EXE_ROOT})
+  IF(WIN32)
+    ADD_TEST(NAME ${TEST_NAME} WORKING_DIRECTORY ${LIBRARY_OUTPUT_PATH}
+      COMMAND ${EXE}${CMAKE_EXECUTABLE_SUFFIX} ${TEST_ARGS} ${${TEST_NAME}_EXTRA_ARGS})
+  ELSE()
+    ADD_TEST(NAME ${TEST_NAME} COMMAND ${EXE} ${TEST_ARGS} ${${TEST_NAME}_EXTRA_ARGS})
+  ENDIF()
+  IF(${TEST_NAME}_SET_RUN_SERIAL)
+    SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES RUN_SERIAL ON)
+  ENDIF()
+  # TriBITS doesn't actually currently support `-D <fullTestName>_ENVIRONMENT`
+  # but we decided to add it anyway
+  IF(${TEST_NAME}_ENVIRONMENT)
+    SET_TESTS_PROPERTIES(${TEST_NAME} PROPERTIES ENVIRONMENT "${${TEST_NAME}_ENVIRONMENT}")
+  ENDIF()
```

Comment:
I don't think I follow the reasoning for adding it.

Reply:
We can discuss to remove it now that we support `_EXTRA_ARGS`.